### PR TITLE
[gdal] New feature "cfitsio"

### PR DESCRIPTION
**Pull request by m-kuhn**
**Describe the pull request**
A new feature "cfitsio" for gdal port

- #### What does your PR fix?  
Refs https://github.com/microsoft/vcpkg/issues/20875

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
all

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
Yes

### CODE REVIEW

**Review by dg0yt — CHANGES_REQUESTED** [review]

**Review by dg0yt — CHANGES_REQUESTED** [review]

**Review by dg0yt — APPROVED** [review]
:+1:

**Comment by m-kuhn** [comment]
Thanks for the support @dg0yt

File: ports/gdal/vcpkg.json [hunk 1]
```diff
@@ -32,6 +31,12 @@
     "zstd"
   ],
   "features": {
+    "cfitsio": {
+      "description": "Enable cfitsio support",
+      "dependencies": [
+        "cfitsio"
+      ]
+    },
```
[the remark this answers is not in the document]
Comment (m-kuhn) [reply ports/gdal/vcpkg.json:39]: For what it's worth I can very well live with cfitsio off by default. I don't remember ever having had my hands on a dataset requiring this.

**Review by NancyLi1013 — APPROVED** [review]

**Comment by m-kuhn** [comment]
I took the opportunity to add another feature "hdf5", integrated into the same PR to avoid merge conflicts requiring manual action due to versioning

File: ports/gdal/vcpkg.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (m-kuhn) [reply ports/gdal/vcpkg.json:38]: For reference, the new feature "hdf5" doesn't change this discussion for me but may change it for others.

**Review by dg0yt — CHANGES_REQUESTED** [review]
I would prefer to not add an "hdf5" feature now. hdf5 is not enabled in static linkage on windows, so the feature will not deliver what it promises. I'm handling that as soon as the vcpkg team manages to review and merge my other PRs (pkgconfig for nmake in proj, libspatialite, etc.)

**Comment by dg0yt** [comment]
I would also suggest to not change a PR once it has the "info-reviewed" label, because it will trigger a new review cycle.

**Comment by m-kuhn** [comment]
> I would prefer to not add an "hdf5" feature now. hdf5 is not enabled in static linkage on windows, so the feature will not deliver what it promises. I'm handling that as soon as the vcpkg team manages to review and merge my other PRs (pkgconfig for nmake in proj, libspatialite, etc.)

:+1: 

> I would also suggest to not change a PR once it has the "info-reviewed" label, because it will trigger a new review cycle.

I understand that and was considering it too. In my experience for small things like this, the review cycle is normally short, while the time to merge is larger. As I'm currently not that often at a keyboard, the manual interaction required by the vcpkg versioning strategy makes it harder to "fire and forget".

**Review by dg0yt — APPROVED** [review]
Back at approved commit.

**Comment by BillyONeal** [comment]
Thanks for the new feature!

**Pull request merged**